Makefile

```diff
@@ -73,6 +73,10 @@ notcompiler: FORCE
 	@$(MAKE) runtime
 	@$(MAKE) modules
 
+libchplcomp: FORCE
+	@echo "Making the compiler library..."
+	@cd compiler/next && $(MAKE) -f Makefile.help libchplcomp
```

Comment:
You'll make life easier for yourself in the future if you:

- Just do `@cd compiler && $(MAKE) $@` here. `$@` is automatically set and names the rule's target.
- Add a `libchplcomp` rule to `compiler/Makefile` that does `cd $LIBCHPLDIR && $(MAKE) -f Makefile.help $@`. This is the rule invoked by the above.
- Add a definition `LIBCHPLDIR=next` to `compiler/make/Makefile.compiler.head`.

This will let you change the location of the libchpl directory just by changing that `LIBCHPLDIR` variable, instead of having to look all through the makefiles and find all the instances of 'next'.

Reply:
I think these are interesting ideas but I think it's better not to do it yet in this case. One reason is that I'm hoping in the future we swap`compiler/` and `compiler/next` - i.e. we move the existing compiler code into the new directory structure. In that case, the change will be harder than just renaming `next` and it'll be easier if these rules are more straightforward.